src/js/client/reset.js

```diff
@@ -0,0 +1,140 @@
+    };
+
+    gpii.express.couchuser.frontend.reset.displayReceipt = function (that, responseData) {
+        var jsonData = JSON.parse(responseData);
+        if (jsonData && jsonData.ok) {
+            that.applier.change("user", jsonData.user);
+            that.locate("form").hide();
+
+            that.templates.html(that.locate("message"), that.options.templates.success, { message: "You have successfully reset your password." });
+        }
+        else {
+            that.templates.html(that.locate("message"), that.options.templates.error, { message: jsonData.message });
+        }
+    };
+
+    gpii.express.couchuser.frontend.reset.refresh = function (that) {
+        that.templates.replaceWith(that.locate("form"), that.options.templates.form, that.model);
+        that.events.markupLoaded.fire();
+    };
+
+    // We have to do this because templates need to be loaded before we initialize our own code.
+    gpii.express.couchuser.frontend.reset.init = function (that) {
+        that.templates.loadTemplates();
+        that.events.markupLoaded.fire();
+    };
+
+    fluid.defaults("gpii.express.couchuser.frontend.reset", {
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        components: {
+            templates: {
+                type: "gpii.templates.hb.client"
+            }
+        },
+        templates: {
+            success: "success",
+            error:   "common-error",
+            form:    "reset-form"
+        },
+        apiUrl: "/api/user",
+        selectors: {
+            form:     ".reset-form",
+            message:  ".reset-message",
+            viewport: ".reset-viewport",
+            code:     "input[name='code']",
+            confirm:  "input[name='confirm']",
+            password: "input[name='password']"
+        },
+        events: {
+            submit:       "preventable",
+            refresh:      "preventable",
+            markupLoaded: "preventable"
+        },
+        invokers: {
+            submit: {
+                funcName: "gpii.express.couchuser.frontend.reset.submit",
+                args: [ "{that}", "{arguments}.0"]
+            },
+            displayError: {
+                funcName: "gpii.express.couchuser.frontend.reset.displayError",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            displayReceipt: {
+                funcName: "gpii.express.couchuser.frontend.reset.displayReceipt",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            init: {
+                funcName: "{templates}.loadTemplates"
+            }
+        },
+        listeners: {
+            onCreate: [
+                {
+                    funcName: "gpii.express.couchuser.frontend.reset.init",
+                    args:     "{that}"
+                }
+            ],
+            markupLoaded: [
```

Comment:
I  have created a pull request for the handlebars refactor:  https://github.com/GPII/gpii-handlebars/pull/7